src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,224 @@
+    let lastColumnsOffsets = tableCellRefs.map(ref => (ref?.current?.nextSibling as HTMLTableCellElement)?.offsetWidth);
+    lastColumnsOffsets = lastColumnsOffsets.filter(Boolean).reverse();
+    // Calculate cumulative widths of next siblings to get the the total offset of each of the cells
+    lastColumnsOffsets = lastColumnsOffsets
+      .map((elem, index) => lastColumnsOffsets.slice(0, index + 1).reduce((a, b) => a + b))
+      .reverse();
+
+    setCellOffsets({ first: [0, ...firstColumnsOffsets], last: [...lastColumnsOffsets, 0] });
+  }, [tableCellRefs]);
+
+  useLayoutEffect(() => {
+    // Request animation frame to make sure effect runs after the browser's automatic table layout algorithm
+    requestAnimationFrame(() => {
+      updateCellOffsets();
+    });
+  }, [updateCellOffsets]);
+
+  return { cellOffsets, updateCellOffsets };
+};
+
+interface GetStickyClassNamesProps {
+  styles: { [key: string]: string };
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  isStuckToTheLeft: boolean;
+  isStuckToTheRight: boolean;
+}
+
+export function getStickyClassNames({
+  styles,
+  isSticky,
+  isLastStickyLeft,
+  isLastStickyRight,
+  isStuckToTheLeft,
+  isStuckToTheRight,
+}: GetStickyClassNamesProps) {
+  return clsx({
+    [styles['sticky-cell']]: isSticky,
+    [styles['sticky-cell-last-left']]: isStuckToTheLeft && isLastStickyLeft,
+    [styles['sticky-cell-last-right']]: isStuckToTheRight && isLastStickyRight,
+  });
+}
+
+export const useStickyColumns = ({
+  containerWidth,
+  hasSelection,
+  isWrapperScrollable,
+  stickyColumns,
+  tableRefObject,
+  visibleColumnsLength,
+}: StickyColumnParams) => {
+  // Check if there are any sticky columns
+  const noStickyColumns = !stickyColumns || (stickyColumns.first === 0 && stickyColumns.last === 0);
+  const columnsLengthWithSelection = visibleColumnsLength + (hasSelection ? 1 : 0);
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  // Compute left table padding
+  const table = tableRefObject.current;
+  const tableLeftPadding = table ? parseInt(getComputedStyle(table).paddingLeft) : 0;
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lastRightStickyColumnIndex = columnsLengthWithSelection - 1 - last;
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
```

Comment:
Would it make sense to reverse the `cellOffsets.last`? In that case the widths can be taken as:

```
cellOffsets.first[first + (hasSelection ? 1 : 0)]
cellOffsets.last[last]
```

Also, if the arrays are not appended with 0 (`setCellOffsets({ first: [0, ...firstColumnsOffsets], last: [...lastColumnsOffsets, 0] })`), the indexing can be done as:

```
cellOffsets.first[first + (hasSelection ? 1 : 0) - 1]
cellOffsets.last[last - 1]
```

which looks better logical